doc/nrf/glossary.rst

```diff
@@ -19,42 +56,358 @@ Commit tag
+
+GPIOTE
+   General-Purpose Input/Output Tasks and Events
+
+Hardware Flow Control (HWFC)
+   A handshaking mechanism used to prevent an overflow of bytes in modems.
+   It is utilizing two dedicated pins on the RS-232 connector, Request to Send and Clear to Send.
+
+Host Controller Interface (HCI)
+   Standardized communication between the host stack (e.g., a PC or mobile phone OS) and the controller (the Bluetooth IC).
+   This standard allows the host stack or controller IC to be swapped with minimal adaptation.
+
+Human Interface Device (HID)
+   Type of a computer device that interacts directly with, and most often takes input from, humans and may deliver output to humans.
+   The term "HID" most commonly refers to the USB-HID specification.
+   This standard allows the host stack or controller IC to be swapped with minimal adaptation.
+
+Integrated Circuit (IC)
+   A semiconductor chip consisting of fabricated transistors, resistors, and capacitors.
+
+Integrated Development Environment (IDE)
+   A software application that provides facilities for software development.
+
+Internet Control Message Protocol (ICMP)
+   The control protocol of the IP stack that enables the establishment of reachability, routes, and so on.
+   This protocol is an integral part of any IP but is unique as it is not a transport protocol to exchange data between hosts.
+
+Internet Protocol version 4 (IPv4)
+   The fourth version in the development of the Internet Protocol (IP).
+   It is the communications protocol that provides an identification and location system for computers on networks.
+   It routes most traffic on the Internet.
+
+Internet Protocol version 6 (IPv6)
+   The latest version of the Internet Protocol (IP).
+   It is the communications protocol that provides an identification and location system for computers on networks and routes traffic across the Internet.
+
+IPv4 address
+   A numerical label that is used to identify a network interface of a computer or other network node participating in an IPv4 computer network.
+
+IPv6 address
+   An alphanumerical label that is used to identify a network interface of a computer or other network node participating in an IPv6 computer network.
+
+Kconfig
+   A kernel configuration language.
+   It was initially created for the Linux kernel.
+
+Kconfig file
+   A configuration file provided by an application (usually called :file:`prj.conf` ).
+   It describes build-time configuration options, also called symbols, namely application-specific values for one or more kernel configuration options.
+   It also defines how they are grouped into menus and sub-menus, and dependencies between them that determine what configurations are valid.
+   You can use it to configure software support to build into the final image.
+   Examples include whether to add networking support, which drivers are needed by the application, etc.
```

Comment:
This glossary entry is combining a couple things that should probably be separated.

A _Kconfig file_ is technically the `Kconfig` files found in various folders throughout NCS and Zephyr that define the configurations, options, etc. for a module, and are written in the [Kconfig language](https://www.kernel.org/doc/html/latest/kbuild/kconfig-language.html) syntax. Lines 169 and 170 would be relevant for this description.

A _Kconfig fragment_ is a combination of lines 168, 171, and 172. Perhaps recombined into:

```
A configuration file used for building an application image with or without software support from specific Kconfig options.
Examples include whether to add networking support, which drivers are needed by the application, etc.
```

And maybe add a _Kconfig project configuration_ (or just _Project configuration_?) with a brief summary:

```
A Kconfig fragment used to define default Kconfig options for an application (usually called :file:`prj.conf`).
These are foundational options for the application that will always be applied to its built image.
However, they can be overridden by applying an additional Kconfig fragment at build time.
```

Reply:
Fixed.